### PR TITLE
add contributing guide

### PR DESCRIPTION
**Pull request by everettraven**
added CONTRIBUTING.md file and updated README contributing section to reference it.

Signed-off-by: Bryce Palmer <bpalmer@redhat.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# How to contribute
+
+oria-operator is Apache 2.0 licensed and accepts contributions via GitHub pull requests. This document outlines some of the conventions on commit message formatting, contact points for developers, and other resources to help get contributions into oria-operator.
```
Comment (laxmikantbpandhare) [inline CONTRIBUTING.md:3]: `oria-operator` is Apache 2.0 licensed and accepts contributions via GitHub pull requests. This document outlines some of the conventions on commit message formatting, contact points for developers, and other resources to help get contributions into `oria-operator`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+- Fork the repository on GitHub
+- Install Go >= `1.18`. See the [Go installation instructions][go-install] for more information.
+- Install Docker. See the [Docker installation instructions][docker-install] for more information.
+- Install KinD (Kubernetes-in-Docker). See the [KinD installation instructions][kind-install] for more information.
+
+## Reporting bugs and creating issues
+
+Reporting bugs is one of the best ways to contribute. However, a good bug report has some very specific qualities. Please review the following information on reporting bugs and creating issues:
+
+If any part of the oria-operator project has bugs or documentation mistakes, please let us know by opening an issue. We treat bugs and mistakes very seriously and believe no issue is too small. Before creating a bug report, please check that an issue reporting the same problem does not already exist.
```
Comment (laxmikantbpandhare) [inline CONTRIBUTING.md:20]: If any part of the `oria-operator` project has bugs or documentation mistakes, please let us know by opening an issue. We treat bugs and mistakes very seriously and believe no issue is too small. Before creating a bug report, please check that an issue reporting the same problem does not already exist.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+- Make sure commit messages are in the proper format (see below).
+- Push branch with changes to a personal fork of the repository.
+- Submit a pull request to operator-framework/oria-operator.
+- The PR must be approved by an approver and receive a LGTM from a reviewer. Approvers and reviewers can be found in the OWNERS file.
+
+Thanks for contributing!
+
+### Code style
+
+The coding style suggested by the Go community is used in oria-operator. See the [style doc][golang-style-doc] for details.
```
Comment (laxmikantbpandhare) [inline CONTRIBUTING.md:53]: The coding style suggested by the Go community is used in `oria-operator.` See the [style doc][golang-style-doc] for details.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+Please follow this style to make oria-operator easy to review, maintain and develop.
```
Comment (laxmikantbpandhare) [inline CONTRIBUTING.md:55]: Please follow this style to make `oria-operator` easy to review, maintain and develop.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+### Format of the commit message
+
+We follow a rough convention for commit messages that is designed to answer two
```
Comment (laxmikantbpandhare) [inline CONTRIBUTING.md:59]: We follow a rough convention for commit messages that are designed to answer two

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+<why this change was made>
+<BLANK LINE>
+<footer>
+```
+
+The first line is the subject and should be no longer than 70 characters, the second line is always blank, and other lines should be wrapped at 80 characters. This allows the message to be easier to read on GitHub as well as in various git tools.
+
+## Documentation
+
+Most contributions involve some sort of documentation. Currently all documentation resides in the `docs/` folder or a succinct section in the README. When making contributions, it is important to ensure that all documentation is updated as necessary.
```
Comment (laxmikantbpandhare) [inline CONTRIBUTING.md:85]: Most contributions involve some sort of documentation. Currently, all documentation resides in the `docs/` folder or a succinct section in the README. When making contributions, it is important to ensure that all documentation is updated as necessary.

**Review by laxmikantbpandhare — CHANGES_REQUESTED** [review]
small neats. Rest looks good to me.

File: CONTRIBUTING.md (hunk 6 above)
Comment (everettraven) [reply CONTRIBUTING.md]: I *think* this is resolved with commit https://github.com/operator-framework/oria-operator/pull/73/commits/c387d7632b2f440788ea0a5cb39293fb9ddbbea2

File: CONTRIBUTING.md (hunk 4 above)
Comment (everettraven) [reply CONTRIBUTING.md]: I think this is resolved with commit https://github.com/operator-framework/oria-operator/commit/c387d7632b2f440788ea0a5cb39293fb9ddbbea2

File: CONTRIBUTING.md (hunk 3 above)
Comment (everettraven) [reply CONTRIBUTING.md]: I think this is resolved with commit https://github.com/operator-framework/oria-operator/commit/c387d7632b2f440788ea0a5cb39293fb9ddbbea2

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+If any part of the oria-operator project has bugs or documentation mistakes, please let us know by opening an issue. We treat bugs and mistakes very seriously and believe no issue is too small. Before creating a bug report, please check that an issue reporting the same problem does not already exist.
+
+To make the bug report accurate and easy to understand, please try to create bug reports that are:
+
+- Specific. Include as much details as possible: which version, what environment, what configuration, etc.
+
+- Reproducible. Include the steps to reproduce the problem. We understand some issues might be hard to reproduce, please include the steps that might lead to the problem.
+
+- Isolated. Please try to isolate and reproduce the bug with minimum dependencies. It would significantly slow down the speed to fix a bug if too many dependencies are involved in a bug report. Debugging external systems that rely on oria-operator is out of scope, but we are happy to provide guidance in the right direction or help with using oria-operator itself.
```
[the remark this answers is not in the document]
Comment (everettraven) [reply CONTRIBUTING.md]: I think this is resolved with commit https://github.com/operator-framework/oria-operator/commit/c387d7632b2f440788ea0a5cb39293fb9ddbbea2

File: CONTRIBUTING.md (hunk 2 above)
Comment (everettraven) [reply CONTRIBUTING.md]: I think this is resolved with commit https://github.com/operator-framework/oria-operator/commit/c387d7632b2f440788ea0a5cb39293fb9ddbbea2

File: CONTRIBUTING.md (hunk 1 above)
Comment (everettraven) [reply CONTRIBUTING.md]: I think this is resolved with commit https://github.com/operator-framework/oria-operator/commit/c387d7632b2f440788ea0a5cb39293fb9ddbbea2

File: CONTRIBUTING.md (hunk 5 above)
Comment (everettraven) [reply CONTRIBUTING.md]: fixed in commit https://github.com/operator-framework/oria-operator/pull/73/commits/87727253cd9141d23650af5d72edbf17bce590e9

**Review by laxmikantbpandhare — APPROVED** [review]
/lgtm

**Pull request merged**